bin/assigner/src/main.cpp

```diff
@@ -849,24 +1220,41 @@ int main(int argc, char *argv[]) {
         public_input_file_name = vm["public-input"].as<std::string>();
     }
 
-    if (gen_mode.has_assignments()) {
+    if (gen_mode.has_circuit() || gen_mode.has_fast_tbl() || gen_mode.has_slow_tbl()) {
```

Comment:
We should not ask `assignment-table` file name for pure circuit generation mode (even if table somehow generated for calculate used rows)

Reply:
Pure circuit generation mode generates constants and selectors part of the assignment table so assignment table file name is needed. Other question is that it may be more reasonable to use circuit file name for constants and selectors, like that:
circuit_file_name.crct
selectors_circuit_file_name.crct
table_header_circuit_file_name.crct
constants_circuit_file_name.crct
witness_assignment_file_name.tbl
pub_inp_assignment_file_name.tbl

I think I will do in that way if you don't mind